mvc/controllers/areas.rst

```diff
@@ -1,8 +1,132 @@
+#. Right-click the ``Areas`` folder and create a new folder named ``Products``. This folder will house the controllers and views folders for the ``Products`` area.
+#. Right-click the ``Areas/Products`` folder and create a new folder named ``Controllers``. This folder will house all the controllers for the ``Products`` area.
+#. Right-click the ``Areas/Products/Controllers`` folder and add a new MVC controller named HomeController. By default, this controller will be created with a single action - ``Index``.
+#. Right-click the ``Areas/Products`` folder and create a new folder named ``Views``. This folder will house all the views for the ``Products`` area.
+#. Right-click the ``Areas/Products/Views`` folder and create a new folder named ``Home`` to match the controller name.
+#. Right-click the ``Areas/Products/Views/Home`` folder and add a new MVC view with the same name as the ``Home`` action: ``Index``.
+
+At this point, you should have a project folder hierarchy that includes the following:
+
+- Project name
+
+  - Areas
+
+    - Products
+
+      - Controllers
+
+        - HomeController
+
+      - Views
+
+        - Home
+
+          - Index.cshtml
+
+To add the Services and Deals areas, you would simply repeat steps 2-7, replacing "Products" with the name of the area you're defining.
+
+Once you've defined the folder hierarchy, you need to tell MVC that each controller is associated with an area. You do that by decorating the controller name with the ``[Area]`` attribute. Here's what the boiler-plate ``Products/HomeController`` looks like with the attribute added.
+
+.. code-block:: c#
+  :emphasize-lines: 4
+
+  ...
+  namespace MyStore.Areas.Products.Controllers
+  {
+      [Area("Products")]
+      public class HomeController : Controller
+      {
+          // GET: /<controller>/
+          public IActionResult Index()
+          {
+              return View();
+          }
+      }
+  }
+
+The final step is to set up a route definition that works with your newly created areas. The :doc:`routing` article goes into detail about how to create route definitions, including using conventional routes versus attribute routes. In this example, we'll use a conventional route. To do so, simply open the ``startup.cs`` file and modify it by adding the highlighted route definition below.
+
+.. code-block:: c#
+  :emphasize-lines: 4-6
+
+  ...
+  app.UseMvc(routes =>
+  {
+      routes.MapRoute(name: "areaRoute",
+              template: "{area:exists}/{controller}/{action}",
+              defaults: new { controller = "Home", action = "Index" });
+
+      routes.MapRoute(
+          name: "default",
+          template: "{controller=Home}/{action=Index}");
+  });
+
+Now, when the user enters http://<yourApp>/products, the Index action method of the ``HomeController`` in the ``Products`` area will be invoked and its view will be sent to the client for rendering.
```

Comment:
Not sure what "user enters" means here ... say "a client requests" or something?